openapi.yaml

```diff
@@ -639,6 +639,51 @@ paths:
       - filter
 # {% if False %}
       - default
+# {% endif %}
+      x-openapi-router-controller: athenian.api.controllers.filter_controller
+      x-codegen-request-body-name: body
+  /filter/environments:
+    post:
+      operationId: filter_environments
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/FilterEnvironmentsRequest'
+        x-body-name: body
+      responses:
+        200:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FilteredEnvironments'
+          description: Deployment environments that satisfy the specified filters.
+        400:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InvalidRequestError'
+          description: The request is invalid.
+        404:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GenericError'
+          description: The specified account does not exist or the user is not a member.
+        422:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSourceDataError'
+          description: The account hasn't submitted any deployment notifications.
```

Comment:
So you differentiate no envs ever and no envs for the given date range, indeed, that's good!